compiler/src/dmd/mtype.d

```diff
@@ -2488,6 +2488,7 @@ extern (C++) final class TypeFunction : TypeNext
         bool isInOutQual;      /// inout on the qualifier
         bool isCtor;           /// the function is a constructor
         bool isReturnScope;    /// `this` is returned by value
+        bool isRvalue;         /// return reference should be treated as rvalue
```

Comment:
please update `mtype.h`, also "return_ed_ reference should be treated as rvalue?